app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/ContactDetailsConfirmAddressController.scala

```diff
@@ -50,26 +52,51 @@ class ContactDetailsConfirmAddressController @Inject() (
 )(implicit ec: ExecutionContext)
     extends FrontendController(mcc) with Cacheable with I18nSupport {
 
+  //TODO - temporary while we're working out where to get it from
+  val updateSoleTraderAddress
+    : (String, JourneyRequest[AnyContent], HeaderCarrier) => Future[Address] = (_, request, _) =>
+    updateBusinessAddress(
+      Address(addressLine1 = "2 Scala Street",
+              addressLine2 = Some("Soho"),
+              townOrCity = "London",
+              postCode = "W1T 2HN"
+      )
+    )(request)
+
+  val updateUkCompanyAddress
+    : (String, JourneyRequest[AnyContent], HeaderCarrier) => Future[Address] =
+    (journeyId, request, hc) =>
+      incorpIdConnector.getDetails(journeyId)(hc)
+        .flatMap(response => updateBusinessAddress(response.companyAddress.toPptAddress)(request))
+
   def displayPage(): Action[AnyContent] =
     (authenticate andThen journeyAction).async { implicit request =>
       request.registration.incorpJourneyId match {
         case Some(journeyId) =>
-          for {
-            incorporationDetails <- incorpIdConnector.getDetails(journeyId)
-            updatedAddress <- updateBusinessAddress(
-              incorporationDetails.companyAddress.toPptAddress
-            )
-          } yield Ok(
-            page(form().fill(
-                   ConfirmAddress(request.registration.primaryContactDetails.useRegisteredAddress)
-                 ),
-                 updatedAddress
-            )
-          )
+          request.registration.organisationDetails.organisationType match {
+            case Some(UK_COMPANY)  => executeAddressUpdate(journeyId, updateUkCompanyAddress)
+            case Some(SOLE_TRADER) => executeAddressUpdate(journeyId, updateSoleTraderAddress)
+            case _                 => throw new InternalServerException(s"Company type not supported")
+          }
+
         case _ => Future(Redirect(routes.RegistrationController.displayPage()))
       }
     }
 
+  private def executeAddressUpdate(
+    journeyId: String,
+    getAddressFunction: (String, JourneyRequest[AnyContent], HeaderCarrier) => Future[Address]
+  )(implicit request: JourneyRequest[AnyContent]) =
+    for {
+      updatedAddress <- getAddressFunction(journeyId, request, hc)
```

Comment:
Potentially pass the `request` and `hc` as implicits to this method.

Reply:
ok, inlined those methods so I can use implicits